packages/fileeditor-extension/src/commands.ts

```diff
@@ -863,20 +874,36 @@ export namespace Commands {
 
   /**
    * Add the New File command
+   *
+   * Defaults to Text/.txt if file type data is not specified
    */
   export function addCreateNewCommand(
     commands: CommandRegistry,
     browserFactory: IFileBrowserFactory,
     trans: TranslationBundle
   ): void {
     commands.addCommand(CommandIDs.createNew, {
-      label: args =>
-        args['isPalette'] ? trans.__('New Text File') : trans.__('Text File'),
-      caption: trans.__('Create a new text file'),
-      icon: args => (args['isPalette'] ? undefined : textEditorIcon),
+      label: args => {
+        if (args.isPalette) {
+          return trans.__('New %1 File', args.fileTypeName ?? 'Text');
```

Comment:
Thanks for working on this @ohrely 

To be able to translate `Text` and eventually filetypenames, I suggest we something like:
```typescript
let fileTypeName = trans.__(args.fileTypeName) ?? trans.__('Text');
return trans.__('New %1 File', fileTypeName);
```

That way we can scrape the `'Text'` string so we can translate it, and eventually we could grab `args.fileTypeName` from somewhere else so we can also make them translatable.

Cheers!

Reply:
Would it be better to do the `trans.__()` call where the strings are defined? https://github.com/jupyterlab/jupyterlab/pull/10009/files#diff-f24ad7a44fbf08e8016813bcb1a4f9153eb0bfe57719aadb64e807d1d011eb37R193-R218

Or is it better to do the translation call here with whatever strings people might pass in?

Reply:
Yes it is always better to do that. If is it possible then we should do that

Reply:
I switched to enumerating every possible string (three per file type) to make the translation as straightforward as possible.  @goanpeca does this approach make sense?

Reply:
Lovely! That works much better indeed! Thanks 😄